packages/database/src/core/migrations/20250410213322-DatatrakWebInitialSchema-modifies-schema.js

```diff
@@ -0,0 +1,383 @@
+
+    CREATE TABLE survey_screen_component (
+      id                         TEXT NOT NULL PRIMARY KEY,
+      question_id                TEXT NOT NULL REFERENCES question
+        ON UPDATE CASCADE ON DELETE RESTRICT,
+      screen_id                  TEXT NOT NULL REFERENCES survey_screen
+        ON UPDATE CASCADE ON DELETE CASCADE,
+      component_number           DOUBLE PRECISION NOT NULL,
+      answers_enabling_follow_up TEXT[] DEFAULT '{}'::TEXT[],
+      is_follow_up               BOOLEAN DEFAULT FALSE,
+      visibility_criteria        VARCHAR,
+      validation_criteria        VARCHAR,
+      question_label             TEXT,
+      detail_label               TEXT,
+      config                     VARCHAR DEFAULT '{}'::character varying
+    );
+
+    CREATE INDEX survey_screen_component_component_number_idx ON survey_screen_component (component_number);
+
+    CREATE INDEX survey_screen_component_question_id_idx ON survey_screen_component (question_id);
+
+    CREATE INDEX survey_screen_component_screen_id_idx ON survey_screen_component (screen_id);
+
+    CREATE TABLE survey_response (
+      id              TEXT NOT NULL PRIMARY KEY,
+      survey_id       TEXT NOT NULL REFERENCES survey
+        ON UPDATE CASCADE ON DELETE RESTRICT,
+      user_id         TEXT NOT NULL REFERENCES user_account
+        ON UPDATE CASCADE ON DELETE RESTRICT,
+      assessor_name   TEXT NOT NULL,
+      start_time      TIMESTAMP WITH TIME ZONE NOT NULL,
+      end_time        TIMESTAMP WITH TIME ZONE NOT NULL,
+      metadata        TEXT,
+      timezone        TEXT,
+      entity_id       TEXT NOT NULL REFERENCES entity
+        ON UPDATE CASCADE ON DELETE RESTRICT,
+      data_time       TIMESTAMP,
+      outdated        BOOLEAN DEFAULT FALSE
+    );
+
+    CREATE INDEX survey_response_end_time_idx ON survey_response (end_time);
+
+    CREATE INDEX survey_response_start_time_idx ON survey_response (start_time);
+
+    CREATE INDEX survey_response_survey_id_idx ON survey_response (survey_id);
+
+    CREATE INDEX survey_response_user_id_idx ON survey_response (user_id);
+
+    CREATE INDEX survey_response_entity_id_idx ON survey_response (entity_id);
+
+    CREATE INDEX survey_response_outdated_id_idx ON survey_response (outdated);
+
+    CREATE INDEX survey_response_data_time_idx ON survey_response (data_time DESC);
+
+    CREATE TABLE answer (
+      id                 TEXT NOT NULL PRIMARY KEY,
+      type               TEXT NOT NULL,
+      survey_response_id TEXT NOT NULL REFERENCES survey_response
+        ON UPDATE CASCADE ON DELETE CASCADE,
+      question_id        TEXT NOT NULL REFERENCES question,
+      text               TEXT,
+
+      CONSTRAINT answer_survey_response_id_question_id_unique
+        UNIQUE (survey_response_id, question_id)
+    );
+
+    ----------------------------
+  `);
+};
+
+exports.down = function (db) {
+  return null;
+};
+
+exports._meta = {
+  version: 1,
+  targets: ['browser'],
```

Comment:
Only runs in browser, thanks to [this PR](https://github.com/beyondessential/tupaia/pull/6145/files)